l10n_th_account_tax_filing/models/account_tax_filing.py

```diff
@@ -0,0 +1,431 @@
+        inverse_name="filing_id",
+        copy=False,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+
+    @api.model
+    def create(self, vals):
+        # Set name based on sequence
+        if vals.get("name", "Draft") == "Draft":
+            vals["name"] = (
+                self.env["ir.sequence"].next_by_code("account.tax.filing") or "Draft"
+            )
+        return super(AccountTaxFiling, self).create(vals)
+
+    @api.onchange("date_range_id")
+    def _onchange_date_range(self):
+        # Set date_from and date_to based on date_range_id
+        for rec in self:
+            if rec.date_range_id:
+                rec.update(
+                    {
+                        "date_from": rec.date_range_id.date_start,
+                        "date_to": rec.date_range_id.date_end,
+                    }
+                )
+
+    @api.depends("tax_filing_line_ids")
+    def _compute_total_amount(self):
+        for rec in self:
+            amount_from = abs(
+                sum(
+                    rec.tax_filing_line_ids.filtered(
+                        lambda line: line.account_id == rec.account_from_id
+                    ).mapped("balance")
+                )
+            )
+            amount_to = abs(
+                sum(
+                    rec.tax_filing_line_ids.filtered(
+                        lambda line: line.account_id == rec.account_to_id
+                    ).mapped("balance")
+                )
+            )
+            amount_adjust = abs(
+                sum(
+                    rec.tax_filing_line_ids.filtered(
+                        lambda line: line.account_id == rec.account_adjust_id
+                    ).mapped("balance")
+                )
+            )
+            rec.write(
+                {
+                    "amount_from": amount_from,
+                    "amount_to": amount_to,
+                    "amount_adjust": amount_adjust,
+                    "total_amount": (amount_from - amount_to) * -1,
+                }
+            )
+
+    @api.constrains("date_from", "date_to")
+    def check_date_from_to(self):
+        # Check date form can't be after date to
+        for rec in self:
+            if rec.date_from and rec.date_to and rec.date_from > rec.date_to:
+                raise UserError(_("Start Date must not be after End Date"))
+
+    def action_compute_account_tax_filing_line(self):
+        # Clear all lines before recompute (if any)
+        self.tax_filing_line_ids.unlink()
+
+        if self.account_adjust_id:
+            condition = "WHERE aml.account_id IN ({}, {}, {})".format(
+                self.account_from_id.id,
+                self.account_to_id.id,
+                self.account_adjust_id.id,
+            )
+        else:
+            condition = "WHERE aml.account_id IN ({}, {})".format(
+                self.account_from_id.id, self.account_to_id.id
+            )
+        query = """
```

Comment:
```suggestion
        query = """
            SELECT aml.id, aml.account_id, aml.name,
                aml.date, aml.debit, aml.credit, aml.balance
            FROM account_move_line aml
            LEFT JOIN account_tax_filing_line filing_line ON filing_line.move_line_id = aml.id
            LEFT JOIN account_account aa ON aa.id = aml.account_id
            LEFT JOIN account_move am ON am.id = aml.move_id
            WHERE aml.account_id IN %s
                AND aml.date BETWEEN %s AND %s
                AND aml.parent_state = 'posted'
                AND filing_line.id IS NULL
                AND aml.tax_filing_id IS NULL
                AND aml.company_id = %s
            ORDER BY
                CASE aa.id
                    WHEN %s THEN 1
                    WHEN %s THEN 2
                    WHEN %s THEN 3
                END,
                aml.date
            """
```

In query, please order by account (adjust > from > to) and date